### PR TITLE
Add `find-superquadric`

### PR DESCRIPTION
**Pull request by pattacini**
As discussed with @traversaro, this draft PR is a placeholder for considering adding `find-superquadric` to the superbuild.

We still need to handle the dependency from `VTK`.

This PR replaces the "unused" `icub-grasp-demo` w/ the new package in the hidden profile `GRASPING`.

### CODE REVIEW

File: cmake/Buildfind-superquadric.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+find_or_build_package(YARP QUIET)
+find_or_build_package(ICUB QUIET)
+find_or_build_package(ICUBcontrib QUIET)
+
+ycm_ep_helper(find-superquadric TYPE GIT
+                                STYLE GITHUB
+                                REPOSITORY robotology/find-superquadric.git
+                                DEPENDS YARP
+                                        ICUB
+                                        ICUBcontrib
```
Comment (traversaro) [inline cmake/Buildfind-superquadric.cmake:17]: @pattacini can I remove the dependency on `ICUBcontrib`? At a first glance, the repo does not depend on it.

**Comment by traversaro** [comment]
We finally have a vtk related option: https://github.com/robotology/robotology-superbuild/pull/1217 . @pattacini do you want to rebase? Otherwise I can do it myself.

**Comment by pattacini** [comment]
I'll do it right now. Thanks!

**Comment by pattacini** [comment]
Done!

**Comment by traversaro** [comment]
@pattacini Now the editing works, probably because I am not using `github.dev` anymore.  As we expected the compilation is failing on Ubuntu Bionic 18.04 . Anyhow, this is not particularly problematic as we already announced that 18.04 is unsupported in https://github.com/robotology/community/discussions/580, even if we are silently running the CI for it.

**Comment by traversaro** [comment]
Windows build are failing due to:
* https://github.com/robotology/robotology-superbuild/issues/1269
* https://github.com/robotology/robotology-superbuild/issues/1266

However, the build on Windows is working fine:
~~~
2022-09-21T16:09:25.5205958Z   Performing download step (git clone) for 'find-superquadric'
2022-09-21T16:09:25.5832607Z   Cloning into 'find-superquadric'...
2022-09-21T16:09:27.3054770Z   Your branch is up to date with 'origin/master'.
2022-09-21T16:09:27.3055714Z   Already on 'master'
2022-09-21T16:09:27.9140790Z   No patch step for 'find-superquadric'
2022-09-21T16:09:27.9684333Z   Performing update step for 'find-superquadric'
2022-09-21T16:09:28.8267026Z   Current branch master is up to date.
2022-09-21T16:09:29.4058156Z   Performing configure step for 'find-superquadric'
2022-09-21T16:09:29.4276018Z   Not searching for unused variables given on the command line.
2022-09-21T16:09:29.4313016Z   loading initial cache file D:/a/robotology-superbuild/robotology-superbuild/build/src/find-superquadric/CMakeFiles/YCMTmp/find-superquadric-cache-Release.cmake
2022-09-21T16:09:31.1472950Z   -- The C compiler identification is MSVC 19.29.30146.0
2022-09-21T16:09:32.2476797Z   -- The CXX compiler identification is MSVC 19.29.30146.0
2022-09-21T16:09:32.2763849Z   -- Detecting C compiler ABI info
2022-09-21T16:09:33.3053027Z   -- Detecting C compiler ABI info - done
2022-09-21T16:09:33.3078469Z   -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
2022-09-21T16:09:33.3086497Z   -- Detecting C compile features
2022-09-21T16:09:33.3096855Z   -- Detecting C compile features - done
2022-09-21T16:09:33.3197694Z   -- Detecting CXX compiler ABI info
2022-09-21T16:09:34.3520176Z   -- Detecting CXX compiler ABI info - done
2022-09-21T16:09:34.3548323Z   -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
2022-09-21T16:09:34.3554292Z   -- Detecting CXX compile features
2022-09-21T16:09:34.3564264Z   -- Detecting CXX compile features - done
2022-09-21T16:09:34.4305429Z   -- Found YARP: D:/a/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.7.2+12-20220919.1+git2b2bff591")
2022-09-21T16:09:34.4365194Z   -- Found YCM: D:/a/robotology-superbuild/robotology-superbuild/build/install/share/cmake/YCM (found version "0.14.2.4-20220711.2+gitc609e37")
2022-09-21T16:09:34.4899339Z   -- Found YARP: D:/a/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.7.2+12-20220919.1+git2b2bff591")
2022-09-21T16:09:35.0711361Z   -- Found GSL: C:/Miniconda3/envs/test/Library/include (found version "2.7") 
2022-09-21T16:09:35.7072890Z   -- Found IPOPT: C:/Miniconda3/envs/test/Library/lib/ipopt-3.lib  
2022-09-21T16:09:37.3847762Z   -- Found ICUB 1.28.1 (Components: iCubDev ctrlLib skinDynLib iKin iDyn learningMachine perceptiveModels actionPrimitives optimization)
2022-09-21T16:09:38.0038639Z   -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
2022-09-21T16:09:38.7402176Z   -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
2022-09-21T16:09:38.7407848Z   -- Looking for pthread_create in pthreads
2022-09-21T16:09:39.5978726Z   -- Looking for pthread_create in pthreads - not found
2022-09-21T16:09:39.5980062Z   -- Looking for pthread_create in pthread
2022-09-21T16:09:40.4702979Z   -- Looking for pthread_create in pthread - not found
2022-09-21T16:09:40.4721759Z   -- Found Threads: TRUE  
2022-09-21T16:09:40.4964564Z   -- Found GLEW: C:/Miniconda3/envs/test/Library/lib/glew32.lib  
2022-09-21T16:09:40.5884935Z   -- Found OpenGL: opengl32  found components: OpenGL 
2022-09-21T16:09:45.6147412Z   -- Found Python3: C:/Miniconda3/envs/test/python.exe (found suitable version "3.10.6", minimum required is "3.10") found components: Interpreter Development.Module Development.Embed 
2022-09-21T16:09:45.7285485Z   -- Found HDF5: hdf5-shared (found version "1.12.2") found components: C HL 
2022-09-21T16:09:45.8160310Z   -- Found JsonCpp: C:/Miniconda3/envs/test/Library/lib/jsoncpp.lib (found suitable version "1.9.5", minimum required is "0.7.0") 
2022-09-21T16:09:45.8545648Z   -- Found OGG: C:/Miniconda3/envs/test/Library/lib/ogg.lib  
2022-09-21T16:09:45.8688990Z   -- Found THEORA: C:/Miniconda3/envs/test/Library/lib/theora.lib  
2022-09-21T16:09:45.9369643Z   -- Found NetCDF: C:/Miniconda3/envs/test/Library/include (found version "4.8.1") 
2022-09-21T16:09:46.2537175Z   -- Found LibXml2: C:/Miniconda3/envs/test/Library/lib/xml2.lib (found version "2.9.14") 
2022-09-21T16:09:47.6021148Z   -- Found ZLIB: C:/Miniconda3/envs/test/Library/lib/z.lib (found version "1.2.12") 
2022-09-21T16:09:47.6226310Z   -- Found GL2PS: C:/Miniconda3/envs/test/Library/lib/gl2ps.lib (found suitable version "1.4.2", minimum required is "1.4.2") 
2022-09-21T16:09:47.8204209Z   -- Found PNG: C:/Miniconda3/envs/test/Library/lib/libpng.lib (found version "1.6.38") 
2022-09-21T16:09:47.8892870Z   -- Found SQLite3: C:/Miniconda3/envs/test/Library/include (found version "3.39.3") 
2022-09-21T16:09:47.9258000Z   -- Found LibPROJ: C:/Miniconda3/envs/test/Library/lib/proj.lib (found version "9.0.1") 
2022-09-21T16:09:47.9376416Z   -- Found Eigen3: C:/Miniconda3/envs/test/Library/include/eigen3 (found version "3.4.0") 
2022-09-21T16:09:47.9853329Z   -- Found EXPAT: C:/Miniconda3/envs/test/Library/lib/expat.lib (found version "2.4.9") 
2022-09-21T16:09:48.0192668Z   -- Found double-conversion: C:/Miniconda3/envs/test/Library/lib/double-conversion.lib  
2022-09-21T16:09:48.0458475Z   -- Found LZ4: C:/Miniconda3/envs/test/Library/lib/liblz4.lib (found version "1.9.3") 
2022-09-21T16:09:48.0709215Z   -- Found LZMA: C:/Miniconda3/envs/test/Library/lib/liblzma.lib (found version "5.2.6") 
2022-09-21T16:09:48.0968356Z   -- Found utf8cpp: C:/Miniconda3/envs/test/Library/include  
2022-09-21T16:09:48.4011546Z   -- Found JPEG: C:/Miniconda3/envs/test/Library/lib/jpeg.lib (found version "80") 
2022-09-21T16:09:48.4574943Z   -- Found TIFF: C:/Miniconda3/envs/test/Library/lib/tiff.lib (found version "4.4.0")  
2022-09-21T16:09:48.5543650Z   -- Found Freetype: C:/Miniconda3/envs/test/Library/lib/freetype.lib (found version "2.12.1") 
2022-09-21T16:09:48.7162194Z   -- VTK_VERSION: 9.1.0
2022-09-21T16:09:48.7337725Z   -- Configuring done
2022-09-21T16:09:49.0613938Z   -- Generating done
2022-09-21T16:09:49.0639024Z   -- Build files have been written to: D:/a/robotology-superbuild/robotology-superbuild/build/src/find-superquadric
2022-09-21T16:09:49.1183940Z   Performing build step for 'find-superquadric'
2022-09-21T16:09:49.2911249Z   Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
2022-09-21T16:09:49.2912484Z   Copyright (C) Microsoft Corporation. All rights reserved.
2022-09-21T16:09:49.2913140Z   
2022-09-21T16:09:49.7964114Z     Checking Build System
2022-09-21T16:09:50.0122924Z     Building Custom Rule D:/a/robotology-superbuild/robotology-superbuild/src/find-superquadric/CMakeLists.txt
2022-09-21T16:09:50.1797935Z     nlp.cpp
2022-09-21T16:09:51.1639765Z     main.cpp
2022-09-21T16:09:53.4108098Z     Generating Code...
2022-09-21T16:09:55.5438036Z     find-superquadric.vcxproj -> D:\a\robotology-superbuild\robotology-superbuild\build\src\find-superquadric\Release\find-superquadric.exe
2022-09-21T16:09:55.6078302Z     Building Custom Rule D:/a/robotology-superbuild/robotology-superbuild/src/find-superquadric/CMakeLists.txt
2022-09-21T16:09:55.7011929Z   Performing install step for 'find-superquadric'
2022-09-21T16:09:55.8813593Z   Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
2022-09-21T16:09:55.8814440Z   Copyright (C) Microsoft Corporation. All rights reserved.
2022-09-21T16:09:55.8815053Z   
2022-09-21T16:09:56.5888992Z     find-superquadric.vcxproj -> D:\a\robotology-superbuild\robotology-superbuild\build\src\find-superquadric\Release\find-superquadric.exe
2022-09-21T16:09:56.7954201Z     -- Install configuration: "Release"
2022-09-21T16:09:56.7955316Z     -- Installing: D:/a/robotology-superbuild/robotology-superbuild/build/install/bin/find-superquadric.exe
2022-09-21T16:09:56.8515418Z   Completed 'find-superquadric'
~~~

So we can merge.

**Pull request merged**